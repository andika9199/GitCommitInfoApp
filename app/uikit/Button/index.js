import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import {
  ITEMS_CENTER,
  DEFAULT_TEXT_BOLD,
  WHITE,
  BLACK,
  GRAY_LIGHT,
} from '../../configs/style';

const styles = StyleSheet.create({
  container: {
    ...ITEMS_CENTER,
    width: 150,
    height: 40,
    borderRadius: 4,
  },
  text: {
    ...DEFAULT_TEXT_BOLD,
    color: WHITE,
  },
  bgBlack: {
    backgroundColor: BLACK,
  },
  bgGray: {
    backgroundColor: GRAY_LIGHT,
  },
});

export default function Button(props) {
  const {disabled, onPress, style, title, titleStyle} = props;
  const bgColor = disabled ? styles.bgGray : styles.bgBlack;
  return (
    <TouchableOpacity
      style={[style, bgColor]}
      onPress={onPress}
      disabled={disabled}>
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  disabled: PropTypes.bool,
  onPress: PropTypes.func,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  title: PropTypes.string,
  titleStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

Button.defaultProps = {
  disabled: false,
  onPress: () => {},
  style: styles.container,
  title: 'Button',
  titleStyle: styles.text,
};
