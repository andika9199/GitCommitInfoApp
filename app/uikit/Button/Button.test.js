import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

it('Button successfully rendered', () => {
  const component = renderer.create(<Button />);
  expect(component).toMatchSnapshot();
});

it('Button successfully rendered but disabled', () => {
  const props = {
    disabled: true,
  };
  const component = renderer.create(<Button {...props} />);
  expect(component).toMatchSnapshot();
});
