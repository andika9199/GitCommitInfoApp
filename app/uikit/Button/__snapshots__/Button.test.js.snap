// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button successfully rendered 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 150,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "Robot",
        "fontSize": 13,
        "fontWeight": "bold",
      }
    }
  >
    Button
  </Text>
</View>
`;

exports[`Button successfully rendered but disabled 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#cccccc",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 150,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "Robot",
        "fontSize": 13,
        "fontWeight": "bold",
      }
    }
  >
    Button
  </Text>
</View>
`;
